modules/vba_documents/app/workers/vba_documents/upload_processor.rb

```diff
@@ -57,29 +64,53 @@ def self.cancel!(jid)
+      end
+      StatsD.increment(STATSD_TIMING, tags: ["jid: #{jid}", "guid: #{@upload.guid}", 'step: download_raw_file',
+                                             "raw_file_size: #{tempfile.size}", "time:#{brt.round(5)}}"])
+
       response = nil
       begin
-        @upload.update(metadata: @upload.metadata.merge(original_file_metadata(tempfile)))
-
-        validate_payload_size(tempfile)
-
-        parts = VBADocuments::MultipartParser.parse(tempfile.path)
-        inspector = VBADocuments::PDFInspector.new(pdf: parts)
-        @upload.update(uploaded_pdf: inspector.pdf_data)
+        parts, inspector = nil
+        brt = Benchmark.realtime do
+          @upload.update(metadata: @upload.metadata.merge(original_file_metadata(tempfile)))
 
-        # Validations
-        validate_parts(@upload, parts)
-        validate_metadata(parts[META_PART_NAME], submission_version: @upload.metadata['version'].to_i)
-        metadata = perfect_metadata(@upload, parts, timestamp)
+          validate_payload_size(tempfile)
 
-        pdf_validator_options = VBADocuments::DocumentRequestValidator.pdf_validator_options
-        validate_documents(parts, pdf_validator_options)
+          parts = VBADocuments::MultipartParser.parse(tempfile.path)
+          inspector = VBADocuments::PDFInspector.new(pdf: parts)
+          @upload.update(uploaded_pdf: inspector.pdf_data)
+        end
+        StatsD.increment(STATSD_TIMING, tags: ["jid: #{jid}", "guid: #{@upload.guid}", 'step: parse_parts',
+                                               "time:#{brt.round(5)}}"])
+
+        metadata = nil
+        brt = Benchmark.realtime do
+          # Validations
+          validate_parts(@upload, parts)
+          validate_metadata(parts[META_PART_NAME], submission_version: @upload.metadata['version'].to_i)
+          metadata = perfect_metadata(@upload, parts, timestamp)
+
+          pdf_validator_options = VBADocuments::DocumentRequestValidator.pdf_validator_options
+          validate_documents(parts, pdf_validator_options)
+        end
+        StatsD.increment(STATSD_TIMING, tags: ["jid: #{jid}", "guid: #{@upload.guid}", 'step: validate',
+                                               "time:#{brt.round(5)}}"])
 
-        response = submit(metadata, parts)
+        brt = Benchmark.realtime do
+          response = submit(metadata, parts)
+        end
+        StatsD.increment(STATSD_TIMING, tags: ["jid: #{jid}", "guid: #{@upload.guid}", 'step: cm_upload',
+                                               "time:#{brt.round(5)}}"])
 
-        process_response(response)
-        log_submission(@upload, metadata)
+        brt = Benchmark.realtime do
+          process_response(response)
+          log_submission(@upload, metadata)
+        end
+        StatsD.increment(STATSD_TIMING, tags: ["jid: #{jid}", "guid: #{@upload.guid}", 'step: process_resp',
+                                               "time:#{brt.round(5)}}"])
```

Comment:
I noticed that some of your `time` tags have a space between the `:` and value, e.g. `"time: #{brt.round(5)}}"` and others (like this one) don't. I don't know enough about StatsD tags to know if the space makes a difference, but thought I'd point it out in case it could impact discoverability later.

Reply:
Fixed